src/core/cgroup.c

```diff
@@ -3322,9 +3315,19 @@ int manager_setup_cgroup(Manager *m) {
                         log_debug("Unified cgroup hierarchy is located at %s. Controllers are on legacy hierarchies.", path);
                 else
                         log_debug("Using cgroup controller " SYSTEMD_CGROUP_CONTROLLER_LEGACY ". File system hierarchy is at %s.", path);
+
+                /* Always enable hierarchical support if it exists... */
+                (void) cg_set_attribute("memory", "/", "memory.use_hierarchy", "1");
```

Comment:
This is a new addition rather than a refactor, worth at least a mention in the commit message? Or a separate commit

Reply:
ah no it moved from further down, nvm